captum/optim/_utils/models.py

```diff
@@ -151,3 +155,106 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
             self.dilation,
             self.groups,
         )
+
+
+def pad_reflective_a4d(x: torch.Tensor, padding: List[int]) -> torch.Tensor:
+    """
+    Reflective padding for all 4 dimensions of an NCHW tensor
+    """
+
+    assert x.dim() == 4
+    assert len(padding) == 8
+
+    if x.size(0) == 2:
+        assert sum(padding[6:]) == 0
+    if x.size(1) == 2:
+        assert sum(padding[4:6]) == 0
+
+    # Pad batch
+    if x.size(0) > 2:
+        if padding[6] != 0:
+            x = torch.cat([x, x.flip([0])[1 : (padding[6] + 1)]], dim=0)
+        if padding[7] != 0:
+            x = torch.cat([x.flip([0])[-(padding[7] + 1) : -1], x], dim=0)
+
+    elif x.size(0) == 1:
+        P = []
+        if padding[6] != 0:
+            P.append(
+                torch.cat(
+                    [x if (i + 1) % 2 == 0 else x.flip([0]) for i in range(padding[6])]
+                )
+            )
+        P.append(x)
+        if padding[7] != 0:
+            P.append(
+                torch.cat(
+                    [x if (i + 1) % 2 == 0 else x.flip([0]) for i in range(padding[7])]
+                )
+            )
+        x = torch.cat(P)
+
+    # Pad channels
+    if x.size(1) > 2:
+        if padding[4] != 0:
+            x = torch.cat([x, x.flip([1])[:, 1 : (padding[4] + 1)]], dim=1)
+        if padding[5] != 0:
+            x = torch.cat([x.flip([1])[:, -(padding[5] + 1) : -1], x], dim=1)
+
+    elif x.size(1) == 1:
+        P = []
+        if padding[4] != 0:
+            P.append(
+                torch.cat(
+                    [x if (i + 1) % 2 == 0 else x.flip([1]) for i in range(padding[4])],
+                    dim=1,
+                )
+            )
+        P.append(x)
+        if padding[5] != 0:
+            P.append(
+                torch.cat(
+                    [x if (i + 1) % 2 == 0 else x.flip([1]) for i in range(padding[5])],
+                    dim=1,
+                )
+            )
+        x = torch.cat(P, dim=1)
+
+    # Pad height and width
+    x = torch.nn.functional.pad(x, padding[:4], "reflect")
+    return x
+
+
+def collect_activations(
+    model,
+    targets: Union[nn.Module, List[nn.Module]],
+    model_input: torch.Tensor = torch.zeros(1, 3, 224, 224),
+) -> ModuleOutputMapping:
+    """
+    Collect target activations for a model.
+    """
+
+    catch_activ = ActivationCatcher(targets)
+    activ_out = catch_activ(model, model_input)
+    return activ_out
+
+
+class ActivationCatcher(object):

```

Comment:
I forgot to mention: We usually do not extend from object. It applies by default in the latest versions of python.

Reply:
Should I extend from nn.module, or something else?

Reply:
I called it activation catcher because I was thinking about how the hooks 'catch' the input as it passes through them. But I can make it ActivationFetcher if that is a better name!

Reply:
It doesn't quite look like a nn.Module. If we think that the model can be passed only once and reused, we can pass `model` as a constructor argument as well.

Reply:
Okay, I have it no longer extending from anything.